monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -793,11 +803,13 @@ object Task extends TaskInstances {
     *
     * Although the effects are unordered, we ensure the order of results
     * matches the order of the input sequence. Also see [[gatherUnordered]].
-    *
-    * Alias for [[zipList]].
     */
-  def gather[A](in: Seq[Task[A]]): Task[List[A]] =
-    zipList(in)
+  def gather[A, M[X] <: TraversableOnce[X]](in: M[Task[A]])
+                                           (implicit cbf: CanBuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] = {
```

Comment:
Currently I've chosen diverging signatures between `gather` and `gatherUnordered` because forcing the result to be a list might be undesirable in more cases than having a vector. WDYT?